src/tasks/wifi.c

```diff
@@ -451,23 +451,31 @@ static void do_beacon()
 {
         const struct esp8266_ipv4_info* client_ipv4 = get_client_ipv4_info();
         const struct esp8266_ipv4_info* softap_ipv4 = get_ap_ipv4_info();
-        if (!*client_ipv4->address && !*softap_ipv4->address) {
+	if (!*client_ipv4->address && !*softap_ipv4->address) {
                 /* Then don't bother since we don't have any IP addresses */
                 return;
         }
 
+	/*
+	 * Check if the socket is closed.  If so, whine about it then move
+	 * on and try re-opening it.
+	 */
+	if (state.beacon.serial &&
+	    !serial_is_connected(state.beacon.serial)) {
+		pr_warning("Beacon socket closed!  WTF!\r\n");
+		state.beacon.serial = NULL;
+	}
+
         /*
          * If here then we have at least one IP.  Send the beacon.  Now
          * we need a channel to send the beacon on. Lets grab one if we
          * don't have one yet.
          */
         if (NULL == state.beacon.serial) {
-                state.beacon.serial =
```

Comment:
Yes.  I originally put the change here and then moved it out.  So this was some leftover from that.